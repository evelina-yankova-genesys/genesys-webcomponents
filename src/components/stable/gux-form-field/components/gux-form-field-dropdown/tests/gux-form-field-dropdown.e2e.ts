import { E2EPage, newE2EPage } from '@stencil/core/testing';
import {
  newSparkE2EPage,
  a11yCheck
} from '../../../../../../../tests/e2eTestUtils';

const axeExclusions = [];

async function newNonrandomE2EPage({
  html
}: {
  html: string;
}): Promise<E2EPage> {
  const page = await newE2EPage();

  await page.evaluateOnNewDocument(() => {
    Math.random = () => 0.5;
  });
  await page.setContent(html);
  await page.waitForChanges();

  return page;
}

describe('gux-form-field-dropdown', () => {
  describe('#render', () => {
    describe('single select dropdown', () => {
      describe('label-position', () => {
        [
          '',
          'label-position="above"',
          'label-position="beside"',
          'label-position="screenreader"'
        ].forEach((componentAttribute, index) => {
          const html = `
            <gux-form-field-dropdown ${componentAttribute}>
              <gux-dropdown>
                <gux-listbox>
                  <gux-option value="a" disabled>Ant</gux-option>
                  <gux-option value="b">Bat</gux-option>
                  <gux-option value="c">Cat</gux-option>
                </gux-listbox>
              </gux-dropdown>
              <label slot="label">Default</label>
              <span slot="error">This is an error message</span>
            </gux-form-field-dropdown>
          `;

          it(`should render component as expected (${index + 1})`, async () => {
            const page = await newNonrandomE2EPage({ html });
            const element = await page.find('gux-form-field-dropdown');
            const elementShadowDom = await element.find(
              'pierce/.gux-form-field-container'
            );

            expect(element.outerHTML).toMatchSnapshot();
            expect(elementShadowDom).toMatchSnapshot();
          });

          it(`should be accessible (${index + 1})`, async () => {
            const page = await newSparkE2EPage({ html });

            await a11yCheck(page, axeExclusions);
          });
        });
      });

      describe('input attributes', () => {
        ['', 'disabled', 'required'].forEach((inputAttribute, index) => {
          const html = `
          <gux-form-field-dropdown>
            <gux-dropdown ${inputAttribute}>
              <gux-listbox>
                <gux-option value="a" disabled>Ant</gux-option>
                <gux-option value="b">Bat</gux-option>
                <gux-option value="c">Cat</gux-option>
              </gux-listbox>
            </gux-dropdown>
            <label slot="label">This will be read by a screen reader</label>
          </gux-form-field-dropdown>
          `;

          it(`should render component as expected (${index + 1})`, async () => {
            const page = await newNonrandomE2EPage({ html });
            const element = await page.find('gux-form-field-dropdown');
            const elementShadowDom = await element.find(
              'pierce/.gux-form-field-container'
            );

            expect(element.outerHTML).toMatchSnapshot();
            expect(elementShadowDom).toMatchSnapshot();
          });

          it(`should be accessible (${index + 1})`, async () => {
            const page = await newSparkE2EPage({ html });

            await a11yCheck(page, axeExclusions);
          });
        });
      });
    });
    describe('multi select dropdown', () => {
      describe('label-position', () => {
        [
          '',
          'label-position="above"',
          'label-position="beside"',
          'label-position="screenreader"'
        ].forEach((componentAttribute, index) => {
          const html = `
            <gux-form-field-dropdown ${componentAttribute}>
              <gux-dropdown-multi-beta>
                <gux-listbox-multi>
                  <gux-option-multi value="a" disabled>Ant</gux-option-multi>
                  <gux-option-multi value="b">Bat</gux-option-multi>
                  <gux-option-multi value="c">Cat</gux-option-multi>
                </gux-listbox-multi>
              </gux-dropdown-multi-beta>
              <label slot="label">Default</label>
              <span slot="error">This is an error message</span>
            </gux-form-field-dropdown>
          `;

          it(`should render component as expected (${index + 1})`, async () => {
            const page = await newNonrandomE2EPage({ html });
            const element = await page.find('gux-form-field-dropdown');
            const elementShadowDom = await element.find(
              'pierce/.gux-form-field-container'
            );

            expect(element.outerHTML).toMatchSnapshot();
            expect(elementShadowDom).toMatchSnapshot();
          });

          it(`should be accessible (${index + 1})`, async () => {
            const page = await newSparkE2EPage({ html });

            await a11yCheck(page, axeExclusions);
          });
        });
      });
      ['', 'disabled', 'required'].forEach((inputAttribute, index) => {
        const html = `
          <gux-form-field-dropdown>
            <gux-dropdown-multi-beta ${inputAttribute}>
              <gux-listbox-multi>
                <gux-option-multi value="a" disabled>Ant</gux-option-multi>
                <gux-option-multi value="b">Bat</gux-option-multi>
                <gux-option-multi value="c">Cat</gux-option-multi>
              </gux-listbox-multi>
            </gux-dropdown-multi-beta>
            <label slot="label">Default</label>
          </gux-form-field-dropdown>
        `;

        it(`should render component as expected (${index + 1})`, async () => {
          const page = await newNonrandomE2EPage({ html });
          const element = await page.find('gux-form-field-dropdown');
          const elementShadowDom = await element.find(
            'pierce/.gux-form-field-container'
          );

          expect(element.outerHTML).toMatchSnapshot();
          expect(elementShadowDom).toMatchSnapshot();
        });

        it(`should be accessible (${index + 1})`, async () => {
          const page = await newSparkE2EPage({ html });

          await a11yCheck(page, axeExclusions);
        });
      });
    });
  });
});
