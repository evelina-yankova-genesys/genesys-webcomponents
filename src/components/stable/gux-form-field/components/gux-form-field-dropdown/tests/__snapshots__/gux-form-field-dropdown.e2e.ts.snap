// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-form-field-dropdown #render multi select dropdown label-position should render component as expected (1) 1`] = `"<gux-form-field-dropdown hydrated=\\"\\"> <gux-dropdown-multi-beta hydrated=\\"\\"> <gux-listbox-multi id=\\"gux-form-field-input-i\\" aria-describedby=\\"gux-form-field-error-i\\" role=\\"listbox\\" aria-multiselectable=\\"true\\" tabindex=\\"0\\" hydrated=\\"\\"> <gux-option-multi value=\\"a\\" disabled=\\"\\" id=\\"gux-option-multi-i\\" role=\\"option\\" class=\\"gux-disabled\\" aria-disabled=\\"true\\" hydrated=\\"\\"><!----><div class=\\"gux-option\\">Ant</div></gux-option-multi> <gux-option-multi value=\\"b\\" id=\\"gux-option-multi-i\\" role=\\"option\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" hydrated=\\"\\"><!----><div class=\\"gux-option\\">Bat</div></gux-option-multi> <gux-option-multi value=\\"c\\" id=\\"gux-option-multi-i\\" role=\\"option\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" hydrated=\\"\\"><!----><div class=\\"gux-option\\">Cat</div></gux-option-multi> </gux-listbox-multi> </gux-dropdown-multi-beta> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Default</label> <span slot=\\"error\\" id=\\"gux-form-field-error-i\\">This is an error message</span> </gux-form-field-dropdown>"`;

exports[`gux-form-field-dropdown #render multi select dropdown label-position should render component as expected (1) 2`] = `
<div class="gux-above gux-form-field-container">
  <div class="gux-above gux-form-field-label">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input gux-input-error">
      <div class="gux-dropdown-container">
        <slot></slot>
      </div>
    </div>
    <div class="gux-form-field-error gux-show">
      <gux-icon hydrated="" icon-name="alert-warning-octogon"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-dropdown #render multi select dropdown label-position should render component as expected (2) 1`] = `"<gux-form-field-dropdown label-position=\\"above\\" hydrated=\\"\\"> <gux-dropdown-multi-beta hydrated=\\"\\"> <gux-listbox-multi id=\\"gux-form-field-input-i\\" aria-describedby=\\"gux-form-field-error-i\\" role=\\"listbox\\" aria-multiselectable=\\"true\\" tabindex=\\"0\\" hydrated=\\"\\"> <gux-option-multi value=\\"a\\" disabled=\\"\\" id=\\"gux-option-multi-i\\" role=\\"option\\" class=\\"gux-disabled\\" aria-disabled=\\"true\\" hydrated=\\"\\"><!----><div class=\\"gux-option\\">Ant</div></gux-option-multi> <gux-option-multi value=\\"b\\" id=\\"gux-option-multi-i\\" role=\\"option\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" hydrated=\\"\\"><!----><div class=\\"gux-option\\">Bat</div></gux-option-multi> <gux-option-multi value=\\"c\\" id=\\"gux-option-multi-i\\" role=\\"option\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" hydrated=\\"\\"><!----><div class=\\"gux-option\\">Cat</div></gux-option-multi> </gux-listbox-multi> </gux-dropdown-multi-beta> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Default</label> <span slot=\\"error\\" id=\\"gux-form-field-error-i\\">This is an error message</span> </gux-form-field-dropdown>"`;

exports[`gux-form-field-dropdown #render multi select dropdown label-position should render component as expected (2) 2`] = `
<div class="gux-above gux-form-field-container">
  <div class="gux-above gux-form-field-label">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input gux-input-error">
      <div class="gux-dropdown-container">
        <slot></slot>
      </div>
    </div>
    <div class="gux-form-field-error gux-show">
      <gux-icon hydrated="" icon-name="alert-warning-octogon"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-dropdown #render multi select dropdown label-position should render component as expected (3) 1`] = `"<gux-form-field-dropdown label-position=\\"beside\\" hydrated=\\"\\"> <gux-dropdown-multi-beta hydrated=\\"\\"> <gux-listbox-multi id=\\"gux-form-field-input-i\\" aria-describedby=\\"gux-form-field-error-i\\" role=\\"listbox\\" aria-multiselectable=\\"true\\" tabindex=\\"0\\" hydrated=\\"\\"> <gux-option-multi value=\\"a\\" disabled=\\"\\" id=\\"gux-option-multi-i\\" role=\\"option\\" class=\\"gux-disabled\\" aria-disabled=\\"true\\" hydrated=\\"\\"><!----><div class=\\"gux-option\\">Ant</div></gux-option-multi> <gux-option-multi value=\\"b\\" id=\\"gux-option-multi-i\\" role=\\"option\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" hydrated=\\"\\"><!----><div class=\\"gux-option\\">Bat</div></gux-option-multi> <gux-option-multi value=\\"c\\" id=\\"gux-option-multi-i\\" role=\\"option\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" hydrated=\\"\\"><!----><div class=\\"gux-option\\">Cat</div></gux-option-multi> </gux-listbox-multi> </gux-dropdown-multi-beta> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Default</label> <span slot=\\"error\\" id=\\"gux-form-field-error-i\\">This is an error message</span> </gux-form-field-dropdown>"`;

exports[`gux-form-field-dropdown #render multi select dropdown label-position should render component as expected (3) 2`] = `
<div class="gux-beside gux-form-field-container">
  <div class="gux-beside gux-form-field-label">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input gux-input-error">
      <div class="gux-dropdown-container">
        <slot></slot>
      </div>
    </div>
    <div class="gux-form-field-error gux-show">
      <gux-icon hydrated="" icon-name="alert-warning-octogon"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-dropdown #render multi select dropdown label-position should render component as expected (4) 1`] = `"<gux-form-field-dropdown label-position=\\"screenreader\\" hydrated=\\"\\"> <gux-dropdown-multi-beta hydrated=\\"\\"> <gux-listbox-multi id=\\"gux-form-field-input-i\\" aria-describedby=\\"gux-form-field-error-i\\" role=\\"listbox\\" aria-multiselectable=\\"true\\" tabindex=\\"0\\" hydrated=\\"\\"> <gux-option-multi value=\\"a\\" disabled=\\"\\" id=\\"gux-option-multi-i\\" role=\\"option\\" class=\\"gux-disabled\\" aria-disabled=\\"true\\" hydrated=\\"\\"><!----><div class=\\"gux-option\\">Ant</div></gux-option-multi> <gux-option-multi value=\\"b\\" id=\\"gux-option-multi-i\\" role=\\"option\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" hydrated=\\"\\"><!----><div class=\\"gux-option\\">Bat</div></gux-option-multi> <gux-option-multi value=\\"c\\" id=\\"gux-option-multi-i\\" role=\\"option\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" hydrated=\\"\\"><!----><div class=\\"gux-option\\">Cat</div></gux-option-multi> </gux-listbox-multi> </gux-dropdown-multi-beta> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Default</label> <span slot=\\"error\\" id=\\"gux-form-field-error-i\\">This is an error message</span> </gux-form-field-dropdown>"`;

exports[`gux-form-field-dropdown #render multi select dropdown label-position should render component as expected (4) 2`] = `
<div class="gux-form-field-container gux-screenreader">
  <div class="gux-form-field-label gux-screenreader">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input gux-input-error">
      <div class="gux-dropdown-container">
        <slot></slot>
      </div>
    </div>
    <div class="gux-form-field-error gux-show">
      <gux-icon hydrated="" icon-name="alert-warning-octogon"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-dropdown #render multi select dropdown should render component as expected (1) 1`] = `"<gux-form-field-dropdown hydrated=\\"\\"> <gux-dropdown-multi-beta hydrated=\\"\\"> <gux-listbox-multi id=\\"gux-form-field-input-i\\" role=\\"listbox\\" aria-multiselectable=\\"true\\" tabindex=\\"0\\" hydrated=\\"\\"> <gux-option-multi value=\\"a\\" disabled=\\"\\" id=\\"gux-option-multi-i\\" role=\\"option\\" class=\\"gux-disabled\\" aria-disabled=\\"true\\" hydrated=\\"\\"><!----><div class=\\"gux-option\\">Ant</div></gux-option-multi> <gux-option-multi value=\\"b\\" id=\\"gux-option-multi-i\\" role=\\"option\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" hydrated=\\"\\"><!----><div class=\\"gux-option\\">Bat</div></gux-option-multi> <gux-option-multi value=\\"c\\" id=\\"gux-option-multi-i\\" role=\\"option\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" hydrated=\\"\\"><!----><div class=\\"gux-option\\">Cat</div></gux-option-multi> </gux-listbox-multi> </gux-dropdown-multi-beta> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Default</label> </gux-form-field-dropdown>"`;

exports[`gux-form-field-dropdown #render multi select dropdown should render component as expected (1) 2`] = `
<div class="gux-above gux-form-field-container">
  <div class="gux-above gux-form-field-label">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input">
      <div class="gux-dropdown-container">
        <slot></slot>
      </div>
    </div>
    <div class="gux-form-field-error">
      <gux-icon hydrated="" icon-name="alert-warning-octogon"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-dropdown #render multi select dropdown should render component as expected (2) 1`] = `"<gux-form-field-dropdown hydrated=\\"\\"> <gux-dropdown-multi-beta disabled=\\"\\" hydrated=\\"\\"> <gux-listbox-multi id=\\"gux-form-field-input-i\\" role=\\"listbox\\" aria-multiselectable=\\"true\\" tabindex=\\"0\\" hydrated=\\"\\"> <gux-option-multi value=\\"a\\" disabled=\\"\\" id=\\"gux-option-multi-i\\" role=\\"option\\" class=\\"gux-disabled\\" aria-disabled=\\"true\\" hydrated=\\"\\"><!----><div class=\\"gux-option\\">Ant</div></gux-option-multi> <gux-option-multi value=\\"b\\" id=\\"gux-option-multi-i\\" role=\\"option\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" hydrated=\\"\\"><!----><div class=\\"gux-option\\">Bat</div></gux-option-multi> <gux-option-multi value=\\"c\\" id=\\"gux-option-multi-i\\" role=\\"option\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" hydrated=\\"\\"><!----><div class=\\"gux-option\\">Cat</div></gux-option-multi> </gux-listbox-multi> </gux-dropdown-multi-beta> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Default</label> </gux-form-field-dropdown>"`;

exports[`gux-form-field-dropdown #render multi select dropdown should render component as expected (2) 2`] = `
<div class="gux-above gux-form-field-container">
  <div class="gux-above gux-form-field-label">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input">
      <div class="gux-disabled gux-dropdown-container">
        <slot></slot>
      </div>
    </div>
    <div class="gux-form-field-error">
      <gux-icon hydrated="" icon-name="alert-warning-octogon"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-dropdown #render multi select dropdown should render component as expected (3) 1`] = `"<gux-form-field-dropdown hydrated=\\"\\"> <gux-dropdown-multi-beta required=\\"\\" hydrated=\\"\\"> <gux-listbox-multi id=\\"gux-form-field-input-i\\" role=\\"listbox\\" aria-multiselectable=\\"true\\" tabindex=\\"0\\" hydrated=\\"\\"> <gux-option-multi value=\\"a\\" disabled=\\"\\" id=\\"gux-option-multi-i\\" role=\\"option\\" class=\\"gux-disabled\\" aria-disabled=\\"true\\" hydrated=\\"\\"><!----><div class=\\"gux-option\\">Ant</div></gux-option-multi> <gux-option-multi value=\\"b\\" id=\\"gux-option-multi-i\\" role=\\"option\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" hydrated=\\"\\"><!----><div class=\\"gux-option\\">Bat</div></gux-option-multi> <gux-option-multi value=\\"c\\" id=\\"gux-option-multi-i\\" role=\\"option\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" hydrated=\\"\\"><!----><div class=\\"gux-option\\">Cat</div></gux-option-multi> </gux-listbox-multi> </gux-dropdown-multi-beta> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Default</label> </gux-form-field-dropdown>"`;

exports[`gux-form-field-dropdown #render multi select dropdown should render component as expected (3) 2`] = `
<div class="gux-above gux-form-field-container">
  <div class="gux-above gux-form-field-label gux-required">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input">
      <div class="gux-dropdown-container">
        <slot></slot>
      </div>
    </div>
    <div class="gux-form-field-error">
      <gux-icon hydrated="" icon-name="alert-warning-octogon"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-dropdown #render single select dropdown input attributes should render component as expected (1) 1`] = `"<gux-form-field-dropdown hydrated=\\"\\"> <gux-dropdown hydrated=\\"\\"> <gux-listbox id=\\"gux-form-field-input-i\\" role=\\"listbox\\" tabindex=\\"0\\" hydrated=\\"\\"> <gux-option value=\\"a\\" disabled=\\"\\" id=\\"gux-option-i\\" role=\\"option\\" class=\\"gux-disabled\\" aria-disabled=\\"true\\" hydrated=\\"\\"><!---->Ant</gux-option> <gux-option value=\\"b\\" id=\\"gux-option-i\\" role=\\"option\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" hydrated=\\"\\"><!---->Bat</gux-option> <gux-option value=\\"c\\" id=\\"gux-option-i\\" role=\\"option\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" hydrated=\\"\\"><!---->Cat</gux-option> </gux-listbox> </gux-dropdown> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">This will be read by a screen reader</label> </gux-form-field-dropdown>"`;

exports[`gux-form-field-dropdown #render single select dropdown input attributes should render component as expected (1) 2`] = `
<div class="gux-above gux-form-field-container">
  <div class="gux-above gux-form-field-label">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input">
      <div class="gux-dropdown-container">
        <slot></slot>
      </div>
    </div>
    <div class="gux-form-field-error">
      <gux-icon hydrated="" icon-name="alert-warning-octogon"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-dropdown #render single select dropdown input attributes should render component as expected (2) 1`] = `"<gux-form-field-dropdown hydrated=\\"\\"> <gux-dropdown disabled=\\"\\" hydrated=\\"\\"> <gux-listbox id=\\"gux-form-field-input-i\\" role=\\"listbox\\" tabindex=\\"0\\" hydrated=\\"\\"> <gux-option value=\\"a\\" disabled=\\"\\" id=\\"gux-option-i\\" role=\\"option\\" class=\\"gux-disabled\\" aria-disabled=\\"true\\" hydrated=\\"\\"><!---->Ant</gux-option> <gux-option value=\\"b\\" id=\\"gux-option-i\\" role=\\"option\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" hydrated=\\"\\"><!---->Bat</gux-option> <gux-option value=\\"c\\" id=\\"gux-option-i\\" role=\\"option\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" hydrated=\\"\\"><!---->Cat</gux-option> </gux-listbox> </gux-dropdown> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">This will be read by a screen reader</label> </gux-form-field-dropdown>"`;

exports[`gux-form-field-dropdown #render single select dropdown input attributes should render component as expected (2) 2`] = `
<div class="gux-above gux-form-field-container">
  <div class="gux-above gux-form-field-label">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input">
      <div class="gux-disabled gux-dropdown-container">
        <slot></slot>
      </div>
    </div>
    <div class="gux-form-field-error">
      <gux-icon hydrated="" icon-name="alert-warning-octogon"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-dropdown #render single select dropdown input attributes should render component as expected (3) 1`] = `"<gux-form-field-dropdown hydrated=\\"\\"> <gux-dropdown required=\\"\\" hydrated=\\"\\"> <gux-listbox id=\\"gux-form-field-input-i\\" role=\\"listbox\\" tabindex=\\"0\\" hydrated=\\"\\"> <gux-option value=\\"a\\" disabled=\\"\\" id=\\"gux-option-i\\" role=\\"option\\" class=\\"gux-disabled\\" aria-disabled=\\"true\\" hydrated=\\"\\"><!---->Ant</gux-option> <gux-option value=\\"b\\" id=\\"gux-option-i\\" role=\\"option\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" hydrated=\\"\\"><!---->Bat</gux-option> <gux-option value=\\"c\\" id=\\"gux-option-i\\" role=\\"option\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" hydrated=\\"\\"><!---->Cat</gux-option> </gux-listbox> </gux-dropdown> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">This will be read by a screen reader</label> </gux-form-field-dropdown>"`;

exports[`gux-form-field-dropdown #render single select dropdown input attributes should render component as expected (3) 2`] = `
<div class="gux-above gux-form-field-container">
  <div class="gux-above gux-form-field-label gux-required">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input">
      <div class="gux-dropdown-container">
        <slot></slot>
      </div>
    </div>
    <div class="gux-form-field-error">
      <gux-icon hydrated="" icon-name="alert-warning-octogon"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-dropdown #render single select dropdown label-position should render component as expected (1) 1`] = `"<gux-form-field-dropdown hydrated=\\"\\"> <gux-dropdown hydrated=\\"\\"> <gux-listbox id=\\"gux-form-field-input-i\\" aria-describedby=\\"gux-form-field-error-i\\" role=\\"listbox\\" tabindex=\\"0\\" hydrated=\\"\\"> <gux-option value=\\"a\\" disabled=\\"\\" id=\\"gux-option-i\\" role=\\"option\\" class=\\"gux-disabled\\" aria-disabled=\\"true\\" hydrated=\\"\\"><!---->Ant</gux-option> <gux-option value=\\"b\\" id=\\"gux-option-i\\" role=\\"option\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" hydrated=\\"\\"><!---->Bat</gux-option> <gux-option value=\\"c\\" id=\\"gux-option-i\\" role=\\"option\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" hydrated=\\"\\"><!---->Cat</gux-option> </gux-listbox> </gux-dropdown> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Default</label> <span slot=\\"error\\" id=\\"gux-form-field-error-i\\">This is an error message</span> </gux-form-field-dropdown>"`;

exports[`gux-form-field-dropdown #render single select dropdown label-position should render component as expected (1) 2`] = `
<div class="gux-above gux-form-field-container">
  <div class="gux-above gux-form-field-label">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input gux-input-error">
      <div class="gux-dropdown-container">
        <slot></slot>
      </div>
    </div>
    <div class="gux-form-field-error gux-show">
      <gux-icon hydrated="" icon-name="alert-warning-octogon"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-dropdown #render single select dropdown label-position should render component as expected (2) 1`] = `"<gux-form-field-dropdown label-position=\\"above\\" hydrated=\\"\\"> <gux-dropdown hydrated=\\"\\"> <gux-listbox id=\\"gux-form-field-input-i\\" aria-describedby=\\"gux-form-field-error-i\\" role=\\"listbox\\" tabindex=\\"0\\" hydrated=\\"\\"> <gux-option value=\\"a\\" disabled=\\"\\" id=\\"gux-option-i\\" role=\\"option\\" class=\\"gux-disabled\\" aria-disabled=\\"true\\" hydrated=\\"\\"><!---->Ant</gux-option> <gux-option value=\\"b\\" id=\\"gux-option-i\\" role=\\"option\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" hydrated=\\"\\"><!---->Bat</gux-option> <gux-option value=\\"c\\" id=\\"gux-option-i\\" role=\\"option\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" hydrated=\\"\\"><!---->Cat</gux-option> </gux-listbox> </gux-dropdown> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Default</label> <span slot=\\"error\\" id=\\"gux-form-field-error-i\\">This is an error message</span> </gux-form-field-dropdown>"`;

exports[`gux-form-field-dropdown #render single select dropdown label-position should render component as expected (2) 2`] = `
<div class="gux-above gux-form-field-container">
  <div class="gux-above gux-form-field-label">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input gux-input-error">
      <div class="gux-dropdown-container">
        <slot></slot>
      </div>
    </div>
    <div class="gux-form-field-error gux-show">
      <gux-icon hydrated="" icon-name="alert-warning-octogon"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-dropdown #render single select dropdown label-position should render component as expected (3) 1`] = `"<gux-form-field-dropdown label-position=\\"beside\\" hydrated=\\"\\"> <gux-dropdown hydrated=\\"\\"> <gux-listbox id=\\"gux-form-field-input-i\\" aria-describedby=\\"gux-form-field-error-i\\" role=\\"listbox\\" tabindex=\\"0\\" hydrated=\\"\\"> <gux-option value=\\"a\\" disabled=\\"\\" id=\\"gux-option-i\\" role=\\"option\\" class=\\"gux-disabled\\" aria-disabled=\\"true\\" hydrated=\\"\\"><!---->Ant</gux-option> <gux-option value=\\"b\\" id=\\"gux-option-i\\" role=\\"option\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" hydrated=\\"\\"><!---->Bat</gux-option> <gux-option value=\\"c\\" id=\\"gux-option-i\\" role=\\"option\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" hydrated=\\"\\"><!---->Cat</gux-option> </gux-listbox> </gux-dropdown> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Default</label> <span slot=\\"error\\" id=\\"gux-form-field-error-i\\">This is an error message</span> </gux-form-field-dropdown>"`;

exports[`gux-form-field-dropdown #render single select dropdown label-position should render component as expected (3) 2`] = `
<div class="gux-beside gux-form-field-container">
  <div class="gux-beside gux-form-field-label">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input gux-input-error">
      <div class="gux-dropdown-container">
        <slot></slot>
      </div>
    </div>
    <div class="gux-form-field-error gux-show">
      <gux-icon hydrated="" icon-name="alert-warning-octogon"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-dropdown #render single select dropdown label-position should render component as expected (4) 1`] = `"<gux-form-field-dropdown label-position=\\"screenreader\\" hydrated=\\"\\"> <gux-dropdown hydrated=\\"\\"> <gux-listbox id=\\"gux-form-field-input-i\\" aria-describedby=\\"gux-form-field-error-i\\" role=\\"listbox\\" tabindex=\\"0\\" hydrated=\\"\\"> <gux-option value=\\"a\\" disabled=\\"\\" id=\\"gux-option-i\\" role=\\"option\\" class=\\"gux-disabled\\" aria-disabled=\\"true\\" hydrated=\\"\\"><!---->Ant</gux-option> <gux-option value=\\"b\\" id=\\"gux-option-i\\" role=\\"option\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" hydrated=\\"\\"><!---->Bat</gux-option> <gux-option value=\\"c\\" id=\\"gux-option-i\\" role=\\"option\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" hydrated=\\"\\"><!---->Cat</gux-option> </gux-listbox> </gux-dropdown> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Default</label> <span slot=\\"error\\" id=\\"gux-form-field-error-i\\">This is an error message</span> </gux-form-field-dropdown>"`;

exports[`gux-form-field-dropdown #render single select dropdown label-position should render component as expected (4) 2`] = `
<div class="gux-form-field-container gux-screenreader">
  <div class="gux-form-field-label gux-screenreader">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input gux-input-error">
      <div class="gux-dropdown-container">
        <slot></slot>
      </div>
    </div>
    <div class="gux-form-field-error gux-show">
      <gux-icon hydrated="" icon-name="alert-warning-octogon"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
  </div>
</div>
`;
